skmultilearn/model_selection/iterative_stratification.py

```diff
@@ -0,0 +1,207 @@
+            return M[0]
+        else:
+            max_val = max(self.desired_samples_per_fold[M])
+            M_prim = np.where(
+                np.array(self.desired_samples_per_fold) == max_val)[0]
+            M_prim = np.array([x for x in M_prim if x in M])
+            return np.random.choice(M_prim, 1)[0]
+
+    def get_most_desired_combination(self):
+        currently_chosen = None
+        currently_best_score = None
+        for combination, evidence in self.samples_with_combination.items():
+            current_score = (len(set(combination)), len(evidence))
+            if current_score[1] == 0:
+                continue
+            if currently_chosen is None:
+                currently_chosen = combination
+                currently_best_score = current_score
+                continue
+
+            if currently_best_score[1] > current_score[1] and currently_best_score[0] < current_score[0]:
+                print(currently_best_score, current_score)
+                currently_chosen = combination
+                currently_best_score = current_score
+
+        if current_score is not None:
+            return currently_chosen
+
+        return None
+
+    def distribute_positive_evidence(self):
+        l = self.get_most_desired_combination()
+        while l is not None:
+            print(l)
+            while len(self.samples_with_combination[l]) > 0:
+                row = self.samples_with_combination[l].pop()
+                print(len(self.samples_with_combination[l]), l, row)
+                if self.rows_used[row]:
+                    continue
+
+                max_val = max(self.desired_samples_per_label_per_fold[l])
+                M = np.where(
+                    np.array(self.desired_samples_per_label_per_fold[l]) == max_val)[0]
+                m = self.label_tie_break(M)
+                self.folds[m].append(row)
+                self.rows_used[row] = True
+                for i in self.per_row_combinations[row]:
+                    if row in self.samples_with_combination[i]:
+                        self.samples_with_combination[i].remove(row)
+                    self.desired_samples_per_label_per_fold[i][m] -= 1
+                self.desired_samples_per_fold[m] -= 1
+
+            l = self.get_most_desired_combination()
+
+    def distribute_negative_evidence(self):
+        self.available_samples = [
+            i for i, v in self.rows_used.items() if not v]
+        self.samples_left = len(self.available_samples)
+
+        while self.samples_left > 0:
+            row = self.available_samples.pop()
+            self.rows_used[row] = True
+            self.samples_left -= 1
+            fold_selected = np.random.choice(
+                np.where(self.desired_samples_per_fold > 0)[0],
+                1)[0]
+            self.desired_samples_per_fold[fold_selected] -= 1
+            self.folds[fold_selected].append(row)
+
+    def _iter_test_indices(self, X, y=None, groups=None):
```

Comment:
Hmmm, please correct me if my understanding is wrong, but I assume that `_iter_test_indices` is like the
main routine that calls all other subroutines such as `distribute_positive_evidence` and `distribute_negative_evidence`.
I believe that the latter two methods will not be called by the user. They're just there to compartmentalize important parts of the algorithm.

My question is, is `_iter_test_indices` a user-facing method? If it's being called from the API, then we should remove the underscore before it. In my opinion, I think the one that should have underscores prepended their names would be all of the `init_*`  methods, the `distrubute_*` methods, and the label tie-in. 

In addition, if `_iter_test_indices` is a user-facing method, then it is necessary to put some docstrings to explain how their parameters.

Reply:
I've made distribute* methods private (start with _), also _iter_test_indices is not a user-facing method, the _iter_test_indices is used internally by the scikit BaseKFold class, the only user-facing method we have here is the init.